experimental/dds/tree/src/SharedTree.ts

```diff
@@ -106,14 +106,36 @@ export type SharedTreeArgs<WF extends WriteFormat = WriteFormat> = [writeFormat:
 export type SharedTreeOptions<
 	WF extends WriteFormat,
 	HistoryCompatibility extends 'Forwards' | 'None' = 'Forwards'
-> = Omit<
-	WF extends WriteFormat.v0_0_2
-		? SharedTreeOptions_0_0_2
-		: WF extends WriteFormat.v0_1_1
-		? SharedTreeOptions_0_1_1
-		: never,
-	HistoryCompatibility extends 'Forwards' ? 'summarizeHistory' : never
->;
+> = SharedTreeBaseOptions &
+	Omit<
+		WF extends WriteFormat.v0_0_2
+			? SharedTreeOptions_0_0_2
+			: WF extends WriteFormat.v0_1_1
+			? SharedTreeOptions_0_1_1
+			: never,
+		HistoryCompatibility extends 'Forwards' ? 'summarizeHistory' : never
+	>;
+
+/**
+ * Configuration options for SharedTree that are independent of write format versions.
+ * @public
+ */
+export interface SharedTreeBaseOptions {
+	/**
+	 * The target number of sequenced edits that the tree will try to store in memory.
+	 * Depending on eviction frequency and the collaboration window, there can be more edits in memory at a given time.
+	 * Edits in the collaboration window are not evicted.
+	 *
+	 * The size is set to infinity by default, meaning that all edits in session are kept within memory.
+	 */
+	inMemoryHistorySize?: number;
+	/**
+	 * The rate a which edits are evicted from memory. This is a factor of the inMemoryHistorySize.
+	 * For example, with the default frequency of 2 and a size of 10, the log will evict once it reaches 20 sequenced edits down to 10 edits,
+	 * also keeping any that are still in the collaboration window.
+	 */
+	editEvictionFrequency?: number;
```

Comment:
This is documented really well. I keep getting tripped up by "frequency" though, it doesn't feel like a frequency. What if we made it an actual frequency, i.e. it defaults to `inMemoryHistorySize * 2` rather than `2`? I think that makes it more clear.

Reply:
Done